lua/nvim-autopairs/completion/handlers.lua

```diff
@@ -15,16 +17,47 @@ M["*"] = function(char, item, bufnr, commit_character)
         or (item.data and item.data.funcParensDisabled)
         or (item.textEdit and item.textEdit.newText and item.textEdit.newText:match "[%(%[%$]")
         or (item.insertText and item.insertText:match "[%(%[%$]")
-        or char == commit_character
+        or vim.tbl_contains(commit_character, char)
     then
         return
     end
 
-    vim.api.nvim_feedkeys(char, "i", true)
+    if vim.tbl_isempty(rules) then
+        return
+    end
+
+    local new_text = ''
+    local add_char = 1
+
+    for _, rule in pairs(rules) do
+        if rule.start_pair then
+            local prev_char, next_char = utils.text_cusor_line(
+                new_text,
+                col + add_char,
+                #rule.start_pair,
+                #rule.end_pair,
+                rule.is_regex
+            )
+            local cond_opt = {
+                ts_node = autopairs.state.ts_node,
+                text = new_text,
+                rule = rule,
+                bufnr = bufnr,
+                col = col + 1,
+                char = char,
+                line = line,
+                prev_char = prev_char,
+                next_char = next_char,
+            }
+            if rule.key_map and rule:can_pair(cond_opt) then
+                vim.api.nvim_feedkeys(rule.key_map, "i", true)
```

Comment:
should have a return here. Maybe user add too many rule for for character "("

Reply:
Done